docs/designers-developers/developers/tutorials/sidebar-tutorial/plugin-sidebar-4-initialize-input.md

```diff
@@ -0,0 +1,110 @@
+# Initialize the input control
+
+Now that the field is available in the editor store, we can surface it to the UI. The first thing we're going to do is to extract our input control to a function so we can expand its functionality and our code stays clear.
+
+```js
+( function( wp ) {
+	var el = wp.element.createElement;
+	var Text = wp.components.TextControl;
+
+	var MetaBlockField = function() {
+		return el( Text, {
+			label: 'Meta Block Field',
+			value: 'Initial value',
+			onChange: function( content ) {
+				console.log( 'content changed to ', content );
+			},
+		});
+	}
+
+	wp.plugins.registerPlugin( 'my-plugin-sidebar', {
+		render: function(){
+			return wp.editPost.PluginSidebar( {
+				name: 'my-plugin-sidebar',
+				icon: 'admin-post',
+				title: 'My plugin sidebar',
+				children: el(
+					'div',
+					{ className: 'sidebar-plugin-content' },
+					el( MetaBlockField )
+				),
+			} );
+		}
+	} );
+} )( window.wp );
+```
+
+Now we can focus solely on the `MetaBlockField` component. Our goal is to initialize it with the value of `sidebar_plugin_meta_block_field`, but also to keep it updated when that value changes.
+
+WordPress has [some utilities to work with data](https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/packages-data/) from the stores. The first we're going to use is [`withSelect`](https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/packages-data/#withselect-mapselecttoprops-function-function), whose signature is:
+
+```js
+withSelect(
+	/* function that takes `select` as input and returns an object containing data */
+)(
+	/* function that takes the previous data as input and returns an UI component */
+);
+```
+
+`withSelect` is useful to pass some data to other components, and update them when the original data changes. Let's use it in our plugin code:
+
+```js
+( function( wp ) {
+	var el = wp.element.createElement;
+	var Text = wp.components.TextControl;
+	var withSelect = wp.data.withSelect;
+
+	// Function that takes `select` as input and returns some data.
+	var selectToData = function( select ) {
```

Comment:
This function is called `mapSelectToProps` in `withSelect` definition.

Reply:
77abb82 and 174fccf